v2.0.1 - Added backwards compatibility framework support for .NET Standard v1.3, and Framework v4.5 / v4.6.2

Latency opened the pull request:
Updated to support .NET Core 2.0.
Updated to support .NET Standard v2.0.
Updated to support .NET Framework v4.7.1.
Consolidated redundant code and expressions.
Repacked for multi-frameworks.
Updated versioning.
Updated test projects.
Signed assembly for GAC and other project integrations.

tonerdo commented:
@Latency Thanks for this, I'd take a look. In the meantime let's keep all indentations Allman Style https://en.wikipedia.org/wiki/Indent_style#Allman_style

Latency commented:
Do I need to repack to code-format or?

You pull it..  and then chek it back in on your end if you have Allman code format styles setup on your end.

 

I use K&R here.  Have been for 20+ yrs since I came from a Linux/UNIX background.

 

 

From: Toni Solarin-Sodara [mailto:notifications@github.com] 
Sent: Tuesday, September 26, 2017 4:38 AM
To: tonerdo/readline <readline@noreply.github.com>
Cc: Latency McLaughlin <latency@bio-hazard.us>; Mention <mention@noreply.github.com>
Subject: Re: [tonerdo/readline] Updated to support .NET Core 2.0, .NET Standard 2.0, and Framework v4.7 (#34)

 

@Latency <https://github.com/latency>  Thanks for this, I'd take a look. In the meantime let's keep all indentations Allman Style https://en.wikipedia.org/wiki/Indent_style#Allman_style

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/tonerdo/readline/pull/34#issuecomment-332171403> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AAwHSGbfs72iTpTCnAbeUGwUtTi4-nWnks5smOImgaJpZM4PjYIe> .  <https://github.com/notifications/beacon/AAwHSANde-D21-6F9U57BW3PB1g9RZ58ks5smOImgaJpZM4PjYIe.gif>

tonerdo commented:
@Latency it should be quite straightforward to change. If you're using any Visual Studio product you can format with some shortcut keys. Allman style is a C# best practice

tonerdo commented:
@Latency still working on this?

Latency commented:
Yes, I will work on it today.  Sorry.. I have been busy.

 

 

 

From: Toni Solarin-Sodara [mailto:notifications@github.com] 
Sent: Saturday, October 7, 2017 10:12 AM
To: tonerdo/readline <readline@noreply.github.com>
Cc: Latency McLaughlin <latency@bio-hazard.us>; Mention <mention@noreply.github.com>
Subject: Re: [tonerdo/readline] Updated to support .NET Core 2.0, .NET Standard 2.0, and Framework v4.7 (#34)

 

@Latency <https://github.com/latency>  still working on this?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/tonerdo/readline/pull/34#issuecomment-334951097> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AAwHSBvkGYMHKem9nC3q4f_jTyYKGttTks5sp7DYgaJpZM4PjYIe> .

Latency commented:
Went through and updated the merge conflicts.

There are some concerns with the MSBuild  automation code I had put in for creating artifacts directory and NuGet packages in the distro.

I managed to create a version / tag release for v1.2.1 w/ NuGet binaries on my forked repo, but ultimately, I feel this should be merged as well.

Thoughts, comments, feedback?

Latency commented:
Bah, somebody is making modifications to 'useHistory'.   Is this new code or stale?   Looking at the timestamps, I can't see any changes for +5 mo.   

Can you tell me what is going on here?  I thought I had the latest check-in from the tracking history.

Latency commented:
Is this request still open?  Updated framework and trying to get this pushed into master.  Please help!

tonerdo commented:
Looking into this

Latency commented:
Closing this PR since you cherry picked my changes into master.

Latency commented:
![readline](https://user-images.githubusercontent.com/788296/45270989-fdeb2280-b457-11e8-8dcc-2d9039d9f58c.png)

Latency commented:
.NET Core 2.0 is deprecated as of September 2018 due to failure to upgrade bug.
Use replacement .NET Core 2.1.
See Microsoft documentation and announcements.

ReadLine.nupkg repacked!!!

Latency commented:
Can we merge this to master please?

Latency commented:
Added target support for .NET Core 3.0 and updated .NET Core 2.1 -> 2.2.

Appveyor does not support the new Core 3.0 run-times yet.  Therefore, appveyor is failing at this time.  Please push through anyway!

Latency commented:
- Changed build configurations to remove v4.6.2 as it is no longer supported by default installation configuration for VS2017-2019.   Reverting to v4.6.1 instead.
- Fixed MSBuild target conditions to exclude non-existent dependencies for target platforms.
- Added build property variable for artifact path directory specifications.
- Adding support for Core 3.0 + .Net 4.8 (pre)

Latency commented:
This is bothering to me.   I tried to run unit tests for XUnit (original) and switched it out to NUnit (current) which yield the identical results.

The problem is that under release test, the Console handle is not being set.   When Console2 class goes to make the Console.X calls... it throws.

I tried setting the stream in to it and during [SetUp] of the test, but it still throws.  Not sure what's going on here.   If anybody cares to look into it further, I will restore the test.ps1 for Appveyor.

---------------------------

This is the new release. for the appveyor reason, I decided to take out target support for Framework 4.8 and Standard 3.0.

The next release I will put them in.. which will be latest and no code differences.

StevenTCramer commented:
@Latency I just cloned your master branch and ran the demo app and history fails.

You don't have "Issues" enabled on your fork so I thought I would respond here.  You can see screencast of the error here. https://www.screencast.com/t/YiIZ01FU7Dgs

Latency commented:
Issues now enabled.  I will take a look into it.

Latency commented:
History fixed!   Let me know if you experience any more issues.

WojciechNagorski commented:
@Latency @tonerdo What is the state of this PR? Are you going to merge it with master and publish to nuget?

Latency commented:
I was wondering the same thing!  @tonerdo needs to pull it.  He is not letting me take it over in his absence.    Still trying to get permissions.

Latency commented:
Patched fix for Issue #59.   Known issue with appveyor not supporting targets for .NET Core 3.0 yet.
Release works just fine.
[Release v2.1.3](https://github.com/Latency/ReadLine/releases/tag/2.1.3)
